src/Streams/FileWriter.cpp

```diff
@@ -1,11 +1,32 @@
 #include "FileWriter.h"
 #include <stdexcept>
+#include <experimental/filesystem>
+namespace fs = std::experimental::filesystem;
 
 namespace Stream
 {
-	FileWriter::FileWriter(const std::string& filename) :
+	std::ios_base::openmode FileWriter::TranslateFlags(const std::string& filename, FileWriter::OpenMode openMode) {
+		if ((openMode & FileWriter::OpenMode::FailIfExist) != 0 && fs::exists(filename)) {
+			throw std::runtime_error("File OpenMode::FailIfExist specified, but file exists: " + filename);
```

Comment:
For a console application, I would probably just want to display the error message to the end user. Using `OpenMode::FailIfExist` in the name would likely not be intuitive to a non C++ programmer. 

Maybe: Writing to non-existing filename " + filename + " failed due to a file with this name already existing"

This comment would also apply to `OpenMode::FailIfNoExist` error message as well.

-Brett

Reply:
You're right, the language should be updated. Mention of the flag should be replaced with a description.